### DISCUSSION TITLE
Speed up full sync with checkpoints

### DISCUSSION THREAD

**Discussion by 0xMelkor** [paradigmxyz/reth#810] [Technical discussions]
### Describe the feature

I'd love Reth could provide a way to download the blockchain snapshot from a repository or CDN, to speed up the full sync. I used checkpoints for consensus layers and they dramatically boost the sync process.

Can we provide the same approach for the Reth execution layer?

### Additional context

This is a directory for the [Ethereum Beacon Chain checkpoints](https://eth-clients.github.io/checkpoint-sync-endpoints/)

**Comment by onbjerg** [comment]
Unfortunately I don't think this will really be feasible. The reason it works for CL is because the total state of a beacon state is like 40 megabytes, so it is possible to essentially just download those 40 megabytes, verify it against a trusted source, and backfill historical data while running.

For execution clients, the state is 100s of gigabytes, and there is no way to really slim it down since you won't know what accounts are going to be in future transactions. The only thing you can really do is what e.g. geth does (or some variation of it), which is serving snapshots of state that you can download in semi-real time: https://github.com/ethereum/devp2p/blob/master/caps/snap.md

Converting it to a discussion so we can discuss snapshot sync methods until there is something actionable :)
